### PR TITLE
Adds demo on unitary designs

### PR DESCRIPTION
**Pull request by glassnotes**
Companion demo to the Haar measure one.

TODO:
 - [x] formal progression from spherical to unitary designs
 - [x] description of Pauli/Clifford group
 - [x] finalize computational experiment examples
 - [x] rewrite experiment using QNode transforms (need to change to transform decorator branch for it to build)
 - [x] conclusions
 - [x] references

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ sympy==1.6.2
 pillow==7.1.0
 pip==20.1
 pyscf==1.7.2
-git+https://github.com/PennyLaneAI/pennylane.git@master
+git+https://github.com/PennyLaneAI/pennylane.git@better-transforms
```
Comment (glassnotes) [inline requirements.txt:13]: Note to self: revert this change once the transforms branch is merged in.

File: demonstrations/tutorial_unitary_designs.py [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+    for op in tape.measurements:
+        op.queue()
+
+######################################################################
+# Now let's run the experiments. For both the noisy and ideal experiments,
+# we create a list of tapes, one for each Clifford. We then execute all
+# the tapes on the device, and compute the fidelity.
+
+# Note: This whole section is a good place for a QNode transform since
+# it's a list of tapes, and we are doing something with the execution results.
```
Comment (josh146) [inline demonstrations/tutorial_unitary_designs.py:489]: Nice! Something like this?

```python

@qml.single_tape_transform
def add_noisy_operation_sequence(tape, damp_factor, depo_factor, flip_prob):
    """Add the noisy operation sequence to a tape prior to measurement"""
    for op in tape.operations:
        op.queue()

    noisy_operation_sequence(damp_factor, depo_factor, flip_prob)

    for m in tape.measurements:
        m.queue()


@qml.single_tape_transform
def return_state(tape):
    """Modify a tape to simply return the quantum state"""
    for op in tape.operations:
        op.queue()

    qml.state()


@qml.qnode_transform
def fidelity_comparison(qnode, damp_factor, depo_factor, flip_prob, single_qubit_cliffords):
    tape = return_state(qnode.qtape)

    ideal_tapes = []
    noisy_tapes = []

    for clifford in single_qubit_cliffords:
        # Set up the ideal tape
        twirled_tape = conjugate_with_clifford(tape, clifford)
        ideal_tapes.append(twirled_tape)

        # Set up the noisy tape
        noisy_tape = add_noisy_operation_sequence(tape, damp_factor, depo_factor, flip_prob)
        twirled_tape = conjugate_with_clifford(noisy_tape, clifford)
        noisy_tapes.append(twirled_tape)

    def fidelity_fn(res):
        # Use the results to compute the fidelities
        fidelities = []
        ideal_results = res[:len(ideal_tapes)]
        noisy_results = res[len(ideal_tapes):]

        for state_1, state_2 in zip(ideal_results, noisy_results):
            fidelities.append(fidelity(state_1[0], state_2[0]))

        return fidelities

    return ideal_tapes + noisy_tapes, fidelity_fn


@qml.qnode(dev)
def test_qnode():
    qml.SX(wires=0)
    return qml.expval(qml.PauliZ(0))

fidelity = fidelity_comparison(test_qnode)
fidelity(damp_factor, depo_factor, flip_prob, single_qubit_cliffords)
```

File: demonstrations/tutorial_haar_measure.py [hunk 3]
```diff
@@ -11,11 +11,13 @@
 
 .. related::
 
+    tutorial_unitary_designs Unitary designs
     quantum_volume Quantum volume
     qsim_beyond_classical Beyond classical computing with qsim
     tutorial_barren_plateaus Barren plateaus
 
 
+
```
Comment (glassnotes) [inline demonstrations/tutorial_haar_measure.py:20]:
```suggestion

```

File: demonstrations/tutorial_unitary_designs.py (hunk 2 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:489]: Yes this is similar to what I was thinking. I played around with it a bit more and, given that we haven't quite ironed out everything with QNode transforms, I've rewritten it to use qfunc transforms instead. The result is really nice because it's a lot more modular, and for the most part eliminates the need for tapes except for within the quantum function transforms explicitly.

Comment (josh146) [reply demonstrations/tutorial_unitary_designs.py]: Oh nice! I just had a look, I was super curious. It seems much more concise!

One question though: how come you made `apply_single_clifford(tape, clifford_string, inverse=False)` a tape transform, could it just be a simple qfunc? e.g., remove the decorator and the unused `tape` argument?

```python
def apply_single_clifford(clifford_string, inverse=False):
    for gate in clifford_string:
        if gate == 'H':
            qml.Hadamard(wires=0)
        else:
            if inverse:
                qml.PhaseShift(np.pi/2, wires=0)
            else:
                qml.PhaseShift(-np.pi/2, wires=0)
```

Comment (josh146) [reply demonstrations/tutorial_unitary_designs.py]: I just deleted my comment because I was looking at the old build version, not the new one 🙈

File: demonstrations/tutorial_unitary_designs.py [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+.. meta::
+    :property="og:description": Learn about designs and their uses in quantum computing.
+
+    :property="og:image": https://pennylane.ai/qml/_images/fano.png
+
+.. related::
+
+    tutorial_haar_measure Understanding the Haar measure
+
+*Author: PennyLane dev team. Posted: XX May 2021. Last updated: XX May 2021.*
```
Comment (glassnotes) [inline demonstrations/tutorial_unitary_designs.py:16]: Note to self: update the date before merging.

File: demonstrations/tutorial_unitary_designs.py [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+
+Since :math:`Y = iXZ`, it is enough to specify Clifford operations by how they
+act on :math:`X` and :math:`Z`.  For a particular Clifford, there are 6 possible
+ways it can transform :math:`X` (:math:`\pm X, \pm Y`, or :math:`\pm Z`).  Once
+that is determined, there are four remaining options for the transformation of
+:math:`Z`, leading to 24 elements total. Expressed in string form, they are:
+
+"""
+
+single_qubit_cliffords = [
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:343]: Wait, I'm not sure I understand this. How exactly do the ways that X and Z can be transformed lead to these products of S and H defining the Clifford group?

File: demonstrations/tutorial_unitary_designs.py [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+ 'H', 'S',
+ 'HS', 'SH', 'SS',
+ 'HSH', 'HSS', 'SHS', 'SSH', 'SSS',
+ 'HSHS', 'HSSH', 'HSSS', 'SHSS', 'SSHS',
+ 'HSHSS', 'HSSHS', 'SHSSH', 'SHSSS', 'SSHSS',
+ 'HSHSSH', 'HSHSSS', 'HSSHSS'
+]
+
+######################################################################
+# .. admonition:: Fun fact
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:354]: This text is great, but I would remove the "fun fact" format and leave it as a normal paragraph

File: demonstrations/tutorial_unitary_designs.py [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+import matplotlib.pyplot as plt
+
+# Scipy allows us to sample Haar-random unitaries directly!
+from scipy.stats import unitary_group
+
+# set the random seed
+np.random.seed(42)
+
+# Use the mixed state simulator
+dev = qml.device("default.mixed", wires=1)
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:385]: :star_struck: I'm always happy to see defaut.mixed being used :grin:

File: demonstrations/tutorial_unitary_designs.py [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+######################################################################
+# We can now apply this transformation to create a noisy version of our ideal
+# quantum function:
+
+# The strengths of various types of noise
+damp_factor = 0.02
+depo_factor = 0.02
+flip_prob = 0.01
+
+noisy_experiment = apply_noise(damp_factor, depo_factor, flip_prob)(ideal_experiment)
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:436]: This is so beautiful! :heart_eyes:

File: demonstrations/tutorial_unitary_designs.py [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+# Now let's repeat the procedure using only Clifford group elements. First, we
+# write a quantum function that performs a Clifford operation based on its
+# string representation.
+
+def apply_single_clifford(clifford_string, inverse=False):
+    for gate in clifford_string:
+        if gate == 'H':
+            qml.Hadamard(wires=0)
+        else:
+            # We must manually apply the inverse as it is not yet supported on default.mixed
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:525]: :crying_cat_face:

File: demonstrations/tutorial_unitary_designs.py [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+# ----------
+#
+# In this demo, we've barely scratched the surface of designs and their
+# applications in quantum computing. While benchmarking is a key application
+# area, there are many others.  The Pauli group as a unitary 1-design has
+# applications in the construction of private quantum channels [#PQC]_.
+# *Approximate* unitary :math:`t`-designs (where the equality in the definition
+# is replaced by approximately equal up to some finite precision) are also of
+# interest, as there ways to construct them that are more efficient than those
+# of exact designs [#Dankert]_. It particular, it has been shown that
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:596]:
```suggestion
# of exact designs [#Dankert]_. In particular, it has been shown that
```

File: demonstrations/tutorial_unitary_designs.py [hunk 11]
```diff
@@ -0,0 +1,692 @@
...
+# approximate complex projective 4-designs have applications to the state
+# distinction problem [#Ambainis]_.
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:598]: State distinction instead of state discrimination? First time I hear this. Is it because discrimination is a loaded word?

File: demonstrations/tutorial_unitary_designs.py [hunk 12]
```diff
@@ -0,0 +1,692 @@
...
+#
+# Furthermore, unitary designs are not the only designs that you'll encounter
+# in quantum computing. The familiar Hadamard gate is just a 2-dimensional
+# example of a broader family of *Hadamard designs*, on which there has been
+# extensive research [#Seberry]_. Some sets of `mutually orthogonal Latin
+# squares <https://en.wikipedia.org/wiki/Mutually_orthogonal_Latin_squares>`__
+# have a direct correspondence with mutually unbiased bases, which are optimal
+# quantum measurements [#Gaeta]_ (as well as complex projective designs); and Latin
+# squares themselves have direct correspondence with affine and projective
+# planes, bringing us full circle back to the Fano plane from which we began.
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:608]: Full triangle :wink:

**Review by ixfoduap — CHANGES_REQUESTED** [review]
Another outstanding tutorial, wow! :star_struck: 

The tutorial could go live as is after correcting a couple of typos. However, I am hoping it can get even better. My main comments are:

- I would like to see a short explanation/example of why designs work. How come we can get an exact answer by evaluating on only a few points?
-  I felt that the flow was broken in different parts, specially after a really, really great introduction. I think the tutorial may be trying to do too much (overly enthusiastic) and could benefit from a bit more focus and compactness
- The images in the beginning are so great! I wish there were more like this throughout

File: demonstrations/tutorial_unitary_designs.py [hunk 13]
```diff
@@ -0,0 +1,692 @@
...
+
+[TODO: picture will be updated with some additional solids]
+
+We see from these illustrations that spherical designs are sets of
+*evenly-spaced points* on the surface of a sphere. As :math:`t` increases, the
+configurations become increasingly sphere-like, which makes intuitive sense.
+Spherical designs exist for all :math:`t` and dimension :math:`d`
+[#Handbook]_. They are not always unique, and may have varying numbers of points
+(as can be seen above). Furthermore, a spherical :math:`t`-design is a
+:math:`k`-design for all :math:`k < t` as well!
```
[the remark this answers is not in the document]
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:129]: I wasn't kidding about the overly-enthusiastic part! :sweat_smile:  But yes good point, I will tone it down :grin:

File: demonstrations/tutorial_unitary_designs.py (hunk 11 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:598]: Ah it's just called "state distinction" in that reference and a couple others that refer to it as an application of 4-designs; I thought maybe it was a slightly different task than state discrimination? I will look into it :+1:

File: demonstrations/tutorial_unitary_designs.py (hunk 12 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:608]: Love it :smiling_imp:

File: demonstrations/tutorial_unitary_designs.py (hunk 8 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:436]: I know :heart_eyes: Being able to use the transforms like this has streamlined the code so much, huge improvement over the original version.

File: demonstrations/tutorial_unitary_designs.py (hunk 5 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:343]: These are just the 24 unique elements generated by H and S. We can see why there are 24 using the combinatorial argument above, but then it takes some work to generate the set. There are two approaches, one of which is to look at all possible ways that X and Z can be mapped to {+/-X, +/- Y, +/-Z}, and determine which Clifford performs each mapping. The other is simpler and is to just multiply together generators until you get 24 unique combinations.  In any case I'll work on improving this explanation :+1:

File: demonstrations/tutorial_unitary_designs.py [hunk 14]
```diff
@@ -0,0 +1,692 @@
...
+
+
```
[the remark this answers is not in the document]
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:131]: That's a good point. I'll work on putting an explicit example of a calculation with a spherical design, I think that will be really helpful for illustrating the point (and also help break up the long amount of text that comes before the main code example).

**Comment by glassnotes** [comment]
> Another outstanding tutorial, wow! star_struck
> 
> The tutorial could go live as is after correcting a couple of typos. However, I am hoping it can get even better. My main comments are:
> 
>     * I would like to see a short explanation/example of why designs work. How come we can get an exact answer by evaluating on only a few points?
> 
>     * I felt that the flow was broken in different parts, specially after a really, really great introduction. I think the tutorial may be trying to do too much (overly enthusiastic) and could benefit from a bit more focus and compactness
> 
>     * The images in the beginning are so great! I wish there were more like this throughout

@ixfoduap ready for re-review. I've made the following major changes to address your comments:
 - added an example for the spherical designs
 - removed the section about complex projective designs (helps the demo get to the point faster, plus I wasn't actually doing anything with them; I've left the idea in one of the callout boxes)
 - added a bit more intuition behind how the Clifford group works and is constructed

In terms of the definitions, it feels too hand-wavy if I just omit them; it's important to show that `=` sign between the finite sum and the integral. But now that I've removed complex projective design, each definition is preceded by some motivation and associated with an explicit example, which hopefully provides enough detail to grasp the concept even if some of the formal math is fuzzy.

File: demonstrations/tutorial_unitary_designs.py [hunk 15]
```diff
@@ -0,0 +1,692 @@
...
+.. figure:: /demonstrations/unitary_designs/shapes.svg
+   :align: center
+   :width: 60%
+
+|
+
+[TODO: picture will be updated with some additional solids]
+
+We see from these illustrations that spherical designs are sets of
```
[the remark this answers is not in the document]
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:129]: Haven't said exactly this, but tried to rework the contents.

File: demonstrations/tutorial_unitary_designs.py [hunk 16]
```diff
@@ -0,0 +1,746 @@
...
+
+cube_average = np.mean([f(*point) for point in cube])
+print(cube_average)
+
+
+######################################################################
+# This clearly differs from the true value. We need a design with :math:`t=4`
+# or better in order to compute this average, and when such a design is
+# available, we may save significant computational time.
+#
```
Comment (glassnotes) [inline demonstrations/tutorial_unitary_designs.py:215]: This could be a nice spot for another graphic that shows a cube and a dodecahedron; the earlier graphic can be modified so that they are swapped out for an octahedron and icosahedron respectively. Allows us to show more examples of shapes without cluttering the original drawing.

File: demonstrations/tutorial_unitary_designs.py [hunk 17]
```diff
@@ -0,0 +1,746 @@
...
+######################################################################
+# We compute the mean and variance of the fidelities, and also plot a histogram.
+
+fid_mean = np.mean(fidelities)
+fid_std = np.std(fidelities)
+
+print(f"Mean fidelity      = {fid_mean}")
+print(f"Std. dev. fidelity = {fid_std}")
+
+plt.hist(fidelities, bins=20)
```
Comment (glassnotes) [inline demonstrations/tutorial_unitary_designs.py:557]: I'm considering to remove this histogram - it's neat to see the distribution, but I don't actually use any results about the distribution save for the average.

File: demonstrations/tutorial_unitary_designs.py [hunk 18]
```diff
@@ -0,0 +1,746 @@
...
+# In this demo, we've barely scratched the surface of designs and their
+# applications in quantum computing. While benchmarking is a key application
+# area, there are many others.  The Pauli group as a unitary 1-design has
+# applications in the construction of private quantum channels [#PQC]_.
+# *Approximate* unitary :math:`t`-designs (where the equality in the definition
+# is replaced by approximately equal up to some finite precision) are also of
+# interest, as there ways to construct them that are more efficient than those
+# of exact designs [#Dankert]_. In particular, it has been shown that
+# approximate complex projective 4-designs have applications to the state
+# distinction problem [#Ambainis]_.
```
Comment (glassnotes) [inline demonstrations/tutorial_unitary_designs.py:652]:
```suggestion
# discrimination problem [#Ambainis]_.
```

File: demonstrations/tutorial_unitary_designs.py [hunk 19]
```diff
@@ -0,0 +1,746 @@
...
+.. meta::
+    :property="og:description": Learn about designs and their uses in quantum computing.
+
+    :property="og:image": https://pennylane.ai/qml/_images/fano.png
+
+.. related::
+
+    tutorial_haar_measure Understanding the Haar measure
+
+*Author: PennyLane dev team. Posted: XX May 2021. Last updated: XX May 2021.*
```
Comment (glassnotes) [inline demonstrations/tutorial_unitary_designs.py:16]:
```suggestion
*Author: PennyLane dev team. Posted: XX June 2021. Last updated: XX June 2021.*
```

File: demonstrations/tutorial_unitary_designs.py [hunk 20]
```diff
@@ -70,20 +70,23 @@
...
-(using the proper measure!), but that would be a lot of parameters to keep track
-of. In fact, it might even be overkill--if the terms in the polynomial have the
-same degree of at most :math:`t`, you can compute the average over the sphere
-using only a small set of points rather than integrating over the entire sphere!
-That set of points is called a spherical :math:`t`-design. More formally [#Handbook]_,
-[#Delsarte]_:
+Before diving into unitary designs, let's look at the sphere for some
+intuition.  Suppose we have a polynomial in :math:`d` variables, and we would
+like to compute its average over the surface of a real, :math:`d`-dimensional
+unit sphere, :math:`S(R^d)`. One can do so by integrating that function over the
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:76]: Consider using "we" instead of "one" for a tutorial.

File: demonstrations/tutorial_unitary_designs.py [hunk 21]
```diff
@@ -70,20 +70,23 @@
...
+sphere (using the proper measure), but that would be a lot of parameters to
+keep track of. 
+
+One could alternatively approximate the average by sampling thousands of points
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:80]: Great paragraph! For me it makes the connection with Monte Carlo integration very clear.

In fact, it makes me think that t-designs are a form of quasi-Monte Carlo, which involve strategies that perform the integration over a fixed set of points, e.g., evenly spaced. The magic here is that they can be chosen so the result is exact

File: demonstrations/tutorial_unitary_designs.py [hunk 22]
```diff
@@ -122,223 +128,266 @@
...
-was perfect, then :math:`\Lambda(|0\rangle \langle 0|) = V|0\rangle \langle
-0|V^\dagger`, and the fidelity is
+configurations become increasingly sphere-like. Looking at this in a different
+way, the more complex a function becomes as its degree increases, the closer the
+:math:`t`-design must be to a sphere; we need to evaluate the function at more
+points in order to gain sufficient information when a function is varying more
+quickly due to a higher degree. In 3 dimensions, a function with degree 2 can
+be approximated with something as simple as a tetrahedron, despite the fact that
+it doesn't look spherical at all, while more complex functions require more
+points and thus more complex configurations for the design.  Spherical designs
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:138]:
```suggestion
points and thus more intricate configurations for the design.  Spherical designs
```
or other synonym to avoid repeated use of complex

File: demonstrations/tutorial_unitary_designs.py [hunk 23]
```diff
@@ -122,223 +128,266 @@
...
-By the definition of designs, this means the Clifford group is also a 1-
-and 2-design.
+import numpy as np
 
-The :math:`n`-qubit Pauli group, :math:`\mathcal{P}(n)`, is the set of all tensor
-products of Pauli operations :math:`X`, :math:`Y`, :math:`Z`, and :math:`I`. The
-:math:`n`-qubit Clifford group, :math:`\mathcal{C}(n)`, is the *normalizer* of the
-Pauli group. In simpler terms, the Clifford group is the set of operations that
-send Paulis to Paulis when acting under conjugation (up to a phase), i.e.,
+# The golden ratio
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:167]: :star_struck:

File: demonstrations/tutorial_unitary_designs.py [hunk 24]
```diff
@@ -122,223 +128,266 @@
...
-   S X S^\dagger = Y, \quad S Y S^\dagger = -X, \quad S Z S^\dagger = Z.
+dodeca_average = np.mean([f(*point) for point in dodecahedron])
+print(dodeca_average)
 
-If both :math:`H` and :math:`S` map Paulis to Paulis, then products of them
-do so as well (in group theory terms, the single-qubit Clifford group is
-generated by :math:`H` and :math:`S`).  For example, consider the action of
-:math:`HS`:
+######################################################################
+# This is exactly the value we expect. What happens if we try to do this using
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:201]: LOOOOOVE this example!

File: demonstrations/tutorial_unitary_designs.py [hunk 25]
```diff
@@ -122,223 +128,266 @@
...
-"""
+######################################################################
+# This clearly differs from the true value. We need a design with :math:`t=4`
+# or better in order to compute this average, and when such a design is
+# available, we may save significant computational time.
+# 
+# Unitary designs
+# ^^^^^^^^^^^^^^^
+# 
+# We've demonstrated how spherical are sets of evenly-spaced points on a sphere
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:219]: Revise this sentence

File: demonstrations/tutorial_unitary_designs.py [hunk 26]
```diff
@@ -122,223 +128,266 @@
...
+#    H X H^\dagger = Z, \quad H Y H^\dagger = -Y, \quad H Z H^\dagger = X.
+#
+# This clearly maps Paulis to Paulis (up to a phase). The other is the phase gate :math:`S`:
+#
+# .. math::
+#
+#    S X S^\dagger = Y, \quad S Y S^\dagger = -X, \quad S Z S^\dagger = Z.
+#
+# If both :math:`H` and :math:`S` map Paulis to Paulis, then products of them
+# do so as well (in group theory terms, the single-qubit Clifford group is
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:373]: Consider putting the sentence in parenthesis as a regular new sentence, without parenthesis

File: demonstrations/tutorial_unitary_designs.py [hunk 27]
```diff
@@ -351,31 +400,36 @@
...
-#    map Paulis with only :math:`X` and :math:`Z` to other Paulis.
+# To see for yourself how this set of unitaries is evenly distributed, try
+# applying each of the Cliffords to the initial state :math:`|0\rangle`, and
+# plot the resulting states on the Bloch sphere. You'll find they are
+# symmetric and evenly spaced; in fact, they are all eigenstates of :math:`X`,
+# :math:`Y`, and :math:`Z`. Furthermore, under the full group action, the result
+# is balanced in the sense that each eigenstate is obtained the same number of
+# times.
+#
+# While we do not explore it here, the multi-qubit Clifford can also be
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:411]:
```suggestion
# The multi-qubit Clifford can also be
```

**Review by ixfoduap — CHANGES_REQUESTED** [review]
I really love the changes!

My only remaining suggestions:
- The tutorial could use more images like the ones appearing in the beginning. Maybe you're working on them already?
- I think you commented that it may be a good idea to remove the plot with the distribution of averages. I agree, I think it's best to just quote the average and standard deviation

**Comment by glassnotes** [comment]
> I really love the changes!
> 
> My only remaining suggestions:
> 
>     * The tutorial could use more images like the ones appearing in the beginning. Maybe you're working on them already?
> 
>     * I think you commented that it may be a good idea to remove the plot with the distribution of averages. I agree, I think it's best to just quote the average and standard deviation

Done, and done! Added two new colourful graphics. Let me know if you think of any other places where one could be nice.

File: requirements.txt (hunk 1 above)
Comment (glassnotes) [inline requirements.txt:13]:
```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

File: demonstrations/tutorial_unitary_designs.py [hunk 28]
```diff
@@ -216,14 +214,14 @@ def f(x, y, z):
...
-# We've demonstrated how spherical are sets of evenly-spaced points on a sphere
-# that you can use as short cuts to evaluate the average of a polynomial up to
-# a given degree :math:`t`. However, there was nothing quantum about this; there
-# weren't even any complex numbers involved. A *unitary design* extends this
-# concept from evenly-distributed points to evenly-distributed unitaries.  More
-# formally, instead of averaging polynomials over spheres, we consider
-# polynomials that are functions of the entries of unitary matrices [#Dankert]_,
-# [#Gross]_.
+# We've learned now that spherical designs are sets of evenly-spaced points, and
+# saw how they can be used as a short cut to evaluate the average of a
```
Comment (ixfoduap) [inline demonstrations/tutorial_unitary_designs.py:218]:
```suggestion
# saw how they can be used as a shortcut to evaluate the average of a
```
I think short cut is used too, but I've always thought of it as a single word

**Review by ixfoduap — APPROVED** [review]
:shipit:

File: demonstrations/tutorial_unitary_designs.py (hunk 28 above)
Comment (glassnotes) [reply demonstrations/tutorial_unitary_designs.py:218]: Oh you're right! :exploding_head:
